src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -565,9 +565,11 @@ public boolean isReferenceTo(PsiElement element) {
       if (chain) return false;
     }
     final PsiElement resolve = element instanceof HaxeComponentName ? resolveToComponentName() : resolve();
+    boolean b = resolve instanceof HaxeNamedComponent;
+    boolean b1 = (resolve instanceof HaxeComponentName || b) &&
+                 resolve.getParent() instanceof HaxeClass;
     if (element instanceof HaxeFile &&
-        resolve instanceof HaxeComponentName &&
-        resolve.getParent() instanceof HaxeClass) {
+        (b1 || resolve instanceof AbstractHaxePsiClass) ) {
```

Comment:
Because element was not a name, we will resolve to the actual entity -- not it's name (via the call to resolve() rather than resolveToParent()).  Therefore we will, as you said, never have a component name when we get to that line, and the test always fails.

The answer isn't to make the test more complex, but to simplify it to:
```    if (element instanceof HaxeFile && resolve instanceof HaxeClass)```
This also makes the tests pass.

In the case of resolving to a ```HaxeFunctionDeclarationWithAttributes```, that is a function so, the reference should not be equated to a file, even though a class is.

This brings up the question about moving between files that aren't named the same as the class is.  (In Java this isn't allowed, but in Haxe it is.)